genologics/lims.py

```diff
@@ -379,21 +380,24 @@ def _get_instances(self, klass, params=dict()):
             root = self.get(node.attrib['uri'], params=params)
         return result
 
-    def get_batch(self, instances):
-        "Get the content of a set of instances using the efficient batch call."
+    def get_batch(self, instances, force=False):
+        """Get the content of a set of instances using the efficient batch call."""
         if not instances:
             return []
-        klass = instances[0].__class__
+
         root = ElementTree.Element(nsmap('ri:links'))
+        klass = None
         result = []
         needs_request=False
         for instance in instances:
-            try:
-                result.append(self.cache[instance.uri])
-            except:
-                needs_request=True
+            if not klass:
```

Comment:
I did the revert, you can open a new PR.

On Tue, Oct 6, 2015 at 2:33 PM, paalmbj <notifications@github.com> wrote:

> In genologics/lims.py
> <https://github.com/SciLifeLab/genologics/pull/125#discussion_r41257258>:
>
> >          result = []
> >          needs_request=False
> >          for instance in instances:
> > -            try:
> > -                result.append(self.cache[instance.uri])
> > -            except:
> > -                needs_request=True
> > +            if not klass:
>
> No, patch is completely broken! Don't merge ;) I'll revert it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/genologics/pull/125/files#r41257258>.
>